src/gui/src/layoutViewer.cpp

```diff
@@ -2203,11 +2228,57 @@ void LayoutViewer::drawPinMarkers(Painter& painter,
             text_anchor_pt.setY(text_anchor_pt.y() - max_dim - text_margin);
           }
 
-          painter.drawString(text_anchor_pt.x(), text_anchor_pt.y(), text_anchor, term->getName());
+          PinText pin_specs;
+          pin_specs.box = box;
+          pin_specs.text = term->getName();
+          pin_specs.pt = text_anchor_pt;
+          pin_specs.anchor = text_anchor;
+          pin_specs.top_bottom = arg_min == 2 || arg_min == 3;
+          // only need bounding box for top and bottom
+          if (pin_specs.top_bottom) {
+            const odb::Rect text_rect = painter.stringBoundaries(pin_specs.pt.x(), pin_specs.pt.y(), pin_specs.anchor, pin_specs.text);
+            text_rect.bloat(text_margin, pin_specs.rect);
+          }
+          pin_text_spec.push_back(pin_specs);
+        }
+      }
+    }
+  }
+
+  for (const auto& pin : pin_text_spec) {
+    bool do_rotate = false;
+    auto anchor = pin.anchor;
+    if (pin.top_bottom) {
+      // check if box will intersect anything any
+      for (const auto& other : pin_text_spec) {
```

Comment:
I'm nervous about an n^2 algorithm in the draw loop.  I don't think we need to be so precise about only rotating the minimal possible set.

Reply:
I tried a few different methods, but ultimately didn't like them. I switched to use an RTree since the query is ~logN (so ~N*logN total) and keep track of the layers in a map to make N smaller, and still allows me to track the boxes to determine if rotation will be needed.